Apps/Sandcastle/gallery/Ground Clamping.html

```diff
@@ -0,0 +1,165 @@
+<div id="toolbar">
+    <div id="terrainMenu"></div>
+    <div id="zoomButtons"></div>
+    <div id="toggleLighting"></div>
+    <div id="sampleButtons"></div>
+</div>
+<script id="cesium_sandcastle_script">
+    function startup(Cesium) {
+        'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer');
+viewer.scene.globe.depthTestAgainstTerrain = true;
+
+var cesiumTerrainProviderMeshes = new Cesium.CesiumTerrainProvider({
+    url : 'https://assets.agi.com/stk-terrain/world',
+    requestWaterMask : true,
+    requestVertexNormals : true
+});
+viewer.terrainProvider = cesiumTerrainProviderMeshes;
+
+Sandcastle.addDefaultToolbarMenu([{
+    //
+    // To clamp a point set the heightReference to CLAMP_TO_GROUND or RELATIVE_TO_GROUND
+    //
+    text : 'Draw Points',
+    onselect : function() {
+        viewer.entities.add({
+            position : Cesium.Cartesian3.fromDegrees(-122.1956, 46.1914),
+            point : {
+                color : Cesium.Color.SKYBLUE,
+                pixelSize : 10,
+                outlineColor : Cesium.Color.YELLOW,
+                outlineWidth : 3,
+                heightReference : Cesium.HeightReference.CLAMP_TO_GROUND
+            }
+        });
+
+        viewer.entities.add({
+            position : Cesium.Cartesian3.fromDegrees(-122.1958, 46.1915),
+            point : {
+                color : Cesium.Color.RED,
+                pixelSize : 10,
+                outlineColor : Cesium.Color.GREEN,
+                outlineWidth : 3,
+                heightReference : Cesium.HeightReference.CLAMP_TO_GROUND
+            }
+        });
+        viewer.zoomTo(viewer.entities);
+    }
+}, {
+    //
+    // To clamp a billboard set the heightReference to CLAMP_TO_GROUND or RELATIVE_TO_GROUND
+    //
+    text : 'Draw Billboard',
+    onselect : function() {
+        viewer.entities.add({
+            position : Cesium.Cartesian3.fromDegrees(-122.1956, 46.1914),
+            billboard : {
+                image : '../images/facility.gif',
+                heightReference : Cesium.HeightReference.CLAMP_TO_GROUND
+            }
+        });
+
+        viewer.zoomTo(viewer.entities);
+    }
+}, {
+    //
+    // A corridor will be clamped automatically if it is filled and has no height or extruded height.
+    // NOTE: Setting height to 0 will disable clamping.
+    //
+    text : 'Draw Corridor',
+    onselect : function() {
+        viewer.entities.add({
+            corridor : {
+                positions : Cesium.Cartesian3.fromDegreesArray([
+                    -122.19, 46.1914,
+                    -122.21, 46.21,
+                    -122.23, 46.21
+                ]),
+                width : 2000.0,
+                material : Cesium.Color.GREEN.withAlpha(0.5)
+            }
+        });
+
+        viewer.zoomTo(viewer.entities);
+    }
+}, {
+    //
+    // A polygon will be clamped automatically if it is filled, has no height or extruded height and
```

Comment:
`filled` is not the best word here since we only support colors currently.  Any non-color material (image, stripes, etc..) will not clamp.